### PR TITLE
feat(neuralsearch): add mode to settings types

### PR DESCRIPTION
**Pull request by cdhawke**
## Motivation

- Update the typing to be able to set the mode on a `setSettings` call.

### CODE REVIEW

**Comment by Haroenv** [comment]
Is this the final wording? I saw `semantic` mentioned elsewhere

**Comment by cdhawke** [comment]
> Is this the final wording? I saw `semantic` mentioned elsewhere

These are the discussion points for the naming. The settled value is `keywordSearch` and `neuralSearch`: https://algolia.atlassian.net/wiki/spaces/NEUR/pages/4414802098/Neuralsearch+API+Design+RFC#Index-Settings
https://algolia.atlassian.net/wiki/spaces/NEUR/pages/4414802098/Neuralsearch+API+Design+RFC#Naming-alternatives-for-the-values-in-the-mode-setting

File: packages/client-search/src/types/Settings.ts
Comment (cdhawke) [reply packages/client-search/src/types/Settings.ts:311]: https://github.com/algolia/AlgoliaWeb/pull/16683

**Review by Fluf22 — COMMENTED** [review]
Fine by me 🆗

**Review by Haroenv — COMMENTED** [review]
looks good, but I don't feel comfortable approving/releasing this when naming can still change

**Comment by cdhawke** [comment]
> looks good, but I don't feel comfortable approving/releasing this when naming can still change

That's fine by me! I added @Nagriar as a reviewer to maybe give us some confirmation on the naming when it's settled

**Review by Nagriar — APPROVED** [review]
looks good!

**Comment by Haroenv** [comment]
I'd wait for @shortcuts, I don't think I have credentials to release anymore

**Review by shortcuts — APPROVED** [review]
all goood thanks!!

**Pull request merged**
